### PR TITLE
Experiment type files

### PR DESCRIPTION
**Pull request by mroderick**
This PR is a work in progress to see if we can generate `.d.ts` files from the JSDoc comments.

#### Background

TypeScript users would like to have better support using Sinon and its libraries

#### Solution

1. Write JSDoc for every public function (this benefits everyone)
1. Generate `.d.ts` file(s) from the JSDoc

#### How to verify

I'm not a TypeScript user, so I don't really know what I am doing here

1. Check out this branch
1. `npm ci`
1. `npm build`
1. `npm link`
1. In another project: `npm link @sinonjs/samsam`
1. Try importing `samsam` into your project and see if the type definitions work as expected (I don't know how to do this)

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+  "include": [
+      "dist/*.js"
```
Comment (mshaaban0) [inline tsconfig.json:3]: do built files contain jsdoc annotations? should this be `lib/**/*.js`?

Comment (mroderick) [reply tsconfig.json:3]: They do.

1. `npm run build`
1. `less dist/samsam.js`
1. Observe JSDoc annotations

File: package.json [hunk 2]
```diff
@@ -6,6 +6,7 @@
   "author": "Christian Johansen",
   "license": "BSD-3-Clause",
   "main": "./lib/samsam",
+  "types": "./dist/samsam/samsam.d.ts",
```
Comment (mshaaban0) [inline package.json:9]: Tested, and works as expected 🎉

Screenshot 2020-11-12 at 12 37 40

I had to change the types path though, it should be `./dist/samsam.d.ts`

Comment (mroderick) [reply package.json]: I've fixed that in a new commit... I think.

Would you mind verifying again?

Comment (mshaaban0) [reply package.json]: LGTM, it works 🚀

**Comment by mroderick** [comment]
This has been released as `@sinonjs/samsam@5.3.0`

**Pull request merged**